packages/api/src/schema/schema.yaml

```diff
@@ -239,6 +241,45 @@ components:
           example: D8321C3E-B29C-45EB-A1BB-A623D8BE0F65
           description:
             ID of object store where to which this stream was recorded
+        pushTargets:
+          type: array
+          example: [ { "id" :"PUSH123", "profile": "720p" } ]
+          description:
+            References to targets where this stream will be simulatenously pushed to
+          items:
+            type: object
+            required: [profile]
+            additionalProperties: false
+            properties:
+              profile:
+                type: string
+                description:
+                  Name of transcoding profile that should be sent. Use "source"
+                  for pushing source stream data
+                minLength: 1
+                maxLength: 500
+                example: aac
+              id:
+                type: string
+                description: ID of push target object where to push this stream
+              inline:
+                type: object
+                description:
+                  Inline push target object. Allowed on stream creation, which
+                  will automate creating the push-targets on the same request.
+                required:
+                  - url
+                additionalProperties: false
+                properties:
```

Comment:
I forget the exact syntax but we should be able to have this be a ref to the pushTarget definition already provided within this schema

Reply:
I spent quite some time trying to make that work, which IMO would be the ideal way as well. Unfortunately, it seems like the way we use `json-schema-to-typescript` and `ajv` ([src](https://github.com/livepeer/livepeer-com/blob/b1402e06389c328cf2d311169f3f3a43d8c1fb4f/packages/api/src/compile-schemas.js#L46)) doesn't really support this, since we compile/validate each schema separately from one another. I tried digging a bit in the lib(s) options to see if there would be some easy way to make it work, but couldn't find one in the end. From what found, we only use `$ref` in our schema from the API paths referencing one of the existing components, but not in between the components themselves.

Then, after realizing that the object inside the `stream` would actually always be different from the full `push-target` object in its own API, since it would only be the "push-target creation object" here and never a full push-target object returned by the API, I thought it'd be good to have only the simpler schema here, with only the `url` and `name` which is all we need on its creation, and I could live in peace with repeating them 😆 

I just had some ideas that I could further explore to try to make the schema `$ref`s work though. I could try digging a bit further on that path if you think it's worth it, and it would probably allow to avoid some other repetitions in the schema as well (and maybe in the future even break it up in multiple files?). Let me know!

Reply:
Ah yeah, that makes sense. We don't need to use Ajv in that way anymore - schema precompilation and whatnot doesn't buy us anything anymore. But yeah, out-of-scope for this project.

> I thought it'd be good to have only the simpler schema here, with only the url and name which is all we need on its creation, and I could live in peace with repeating them 😆

Yup, can totally live with it for now.

Reply:
Cool. Maybe later we can maybe give a try to this lib which is specialized on openapi->TS?
https://github.com/drwpow/openapi-typescript